app/controllers/v3/service_instances_controller.rb

```diff
@@ -394,17 +396,26 @@ def admin?
     permission_queryer.can_write_globally?
   end
 
-  def service_plan_valid?(service_plan, space)
+  def service_plan_valid?(service_plan)
     service_plan &&
-      visible_to_current_user?(plan: service_plan) &&
-      service_plan.visible_in_space?(space)
+      visible_to_current_user?(plan: service_plan)
+  end
+
+  def service_plan_active?(service_plan)
+    service_plan.active?
+  end
+
+  def service_plan_exists_in_space?(service_plan, space)
+    service_plan.visible_in_space?(space)
   end
 
   def raise_if_invalid_service_plan!(service_instance, message)
     return unless message.service_plan_guid
 
     service_plan = ServicePlan.first(guid: message.service_plan_guid)
-    unprocessable_service_plan! unless service_plan_valid?(service_plan, service_instance.space)
+    unprocessable_service_plan! unless service_plan_valid?(service_plan)
+    unavailable_service_plan!(service_plan) unless service_plan_active?(service_plan)
+    service_plan_not_visible_in_space!(service_plan, space) unless service_plan_exists_in_space?(service_plan, service_instance.space)
```

Comment:
`space` is not defined. This leads to a 500 error.
```suggestion
    service_plan_not_visible_in_space!(service_plan, service_instance.space) unless service_plan_exists_in_space?(service_plan, service_instance.space)
```